### PR TITLE
feat(scripts): update IAP subscription check on inactive uids script

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - the script was checking for IAP subscriptions on every account in the db results

This commit:
 - builds a set of account uids that might have IAP subscriptions and only check when the account uid is in the set
 - moved some code into a file that will be shared with another script

### CODE REVIEW

File: packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts [hunk 1]
```diff
@@ -134,88 +129,60 @@ const init = async () => {
...
-    return accessTokens.length > 0;
-  };
+  const sessionTokensFn = fxaDb.sessions.bind(fxaDb);
+  const refreshTokensFn = oauthDb.getRefreshTokensByUid.bind(oauthDb);
+  const accessTokensFn = oauthDb.getAccessTokensByUid.bind(oauthDb);
+
+  const iapSubUids = new Set<string>();
+  const playSubscriptionsCollection = await playBilling.purchaseDbRef().get();
+  const appleSubscriptionsCollection = await appleIap.purchasesDbRef().get();
+  ((collections) => {
```
Comment (vbudhram) [inline packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts:156]: 👍🏽

**Review by vbudhram — APPROVED** [review]

File: packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts [hunk 2]
```diff
@@ -134,88 +129,60 @@ const init = async () => {
...
+    for (const c of collections) {
+      for (const purchaseRecordSnapshot of c.docs) {
+        const x = purchaseRecordSnapshot.data();
```
[the remark this answers is not in the document]
Comment (chenba) [reply packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts:159]: I'm not too worried about the extra check here and there.

[the remark this answers is not in the document]
Comment (StaberindeZA) [reply packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts:159]: Should an inactive subscription block an account delete though?

[the remark this answers is not in the document]
Comment (StaberindeZA) [reply packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts:159]: Assuming that is what's happening. I haven't gone through the code in detail.

**Review by StaberindeZA — APPROVED** [review]

File: packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (chenba) [reply packages/fxa-auth-server/scripts/delete-inactive-accounts/get-inactive-account-uids.ts:159]: It's not blocking the delete.  The logic is, was this account potentially ever an IAP subscriber, if so then perform the normal checking against Firestore.  This allows us to skip 99+% of the accounts.  And I don't want the risks of duplicating or refactoring out the IAP sub logic in the user managers.

**Pull request merged**